Warehouse.Persistence.EF/Persistence/Repositories/BasketRepository.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Warehouse.Application.Common.Interfaces.Persistence;
+using Warehouse.Domain.Entities.Baskets;
+using Warehouse.Domain.Exceptions.BasketExceptions;
+using Warehouse.Persistence.Abstractions;
+using Warehouse.Persistence.EF.Persistence.Contexts;
+
+namespace Warehouse.Persistence.EF.Persistence.Repositories;
+
+public class BasketRepository : GenericRepository<Basket>, IBasketRepository
+{
+    public BasketRepository(WarehouseDbContext dbContext) : base(dbContext)
+    {
+    }
+
+    public async Task<Basket> GetSingleBasketByUserIdAsync(Guid userId)
+    {
+        try
+        {
+            return await _dbContext.Set<Basket>()
+                .Include(b => b.BasketLines)
+                .SingleAsync(o => o.UserId == userId);
+        }
+        catch (InvalidOperationException ex)
```

Comment:
you need  to catch and genaral exeption also. Not onfly for the single and in this case sould return 500 or something diffrent